authentik/core/api/users.py

```diff
@@ -453,8 +455,35 @@ def get_ql_fields(self):
 
     def get_queryset(self):
         base_qs = User.objects.all().exclude_anonymous()
+
+        # Annotate is_superuser to avoid expensive recursive CTE for each user
+        # This checks if user is in any superuser group
+        base_qs = base_qs.annotate(
+            is_superuser=Exists(
+                Group.objects.filter(
+                    users=OuterRef("pk"),
+                    is_superuser=True,
+                )
+            )
+        )
```

Comment:
This can cause wrong data to be shown to the user, as the user might be superuser through an indirect membership. cc. @BeryJu